Organizing compare-like functions

castulo opened the pull request:
We used compare-like functions as arguments to other functions that
filter, sort, compare data. These functions are scattered accros the
code and they are sometimes difficult to identify so there is a lot of
code duplication.

This commit organizes all the compare like functions throughout the code
in the same place and with standard naming so it is easier to find them
and reuse them.

Closes #1167

Signed-off-by: Castulo Martinez <castulo.martinez@intel.com>

otaviobp commented on src/swupd.h:
```diff
@@ -409,6 +405,27 @@ enum swupd_code check_update();
 
+/* swupd_comp_funtions.c */
+
+int find_manifest(const void *a, const void *b);
```
It makes more sense to group all this functions on a new header file instead of just adding it here.
Grouping all functions is a single header is not a good approach, so we should try to move functions from swupd.h and not to group them here.

castulo replied:
I did consider having all these  in `src/swupd_comp_functions.h`, then just having `swupd.h` include `src/swupd_comp_functions.h`, that way all files that needed a comparison function will work as they were. But the problem I encountered was that the comparison functions needed things that are defined in `swupd.h`, like the file or manifest structs, and I didn't want to cause a circular reference in the header files. So I decided to go with this approach so we didn't have to specifically include `src/swupd_comp_functions.h` in all files that needed a function from there. But if you prefer it that way @otaviobp I can change that back to have their own `src/swupd_comp_functions.h` header file.

otaviobp replied:
There's no struct used on any function definition, so you can create a new header file for them without including swupd.h

otaviobp commented on src/swupd.h:
```diff
@@ -409,6 +405,27 @@ enum swupd_code check_update();
+int cmp_manifest_component(const void *a, const void *b);
+int cmp_subscription_component(const void *a, const void *b);
+
+int sort_fullfile(const void *a, const void *b);
```
The name of this function is not good. The function doesn't sort anything, it's a compare function.
We could have a better naming standard for cmp functions.
Maybe something like: `cmp_ + type of first parameter + type of second parameter + some other sufix.`
So in this case `cmp_file_file`.

otaviobp replied:
The same applies to all find_, sort_, filter_ functions here

castulo commented:
V2:

- Removed the categories and renamed all the functions following the same name convention
- Returned functions that were too specific (and unlikely to reuse) as static functions in the file where they are used
- Moved all functions in their own header file instead of using `swupd.h`
